pkg/services/folder/folderimpl/sqlstore.go

```diff
@@ -342,27 +346,155 @@ func (ss *sqlStore) GetHeight(ctx context.Context, foldrUID string, orgID int64,
 	return height, nil
 }
 
-func (ss *sqlStore) GetFolders(ctx context.Context, orgID int64, uids []string) ([]*folder.Folder, error) {
-	if len(uids) == 0 {
-		return []*folder.Folder{}, nil
+// GetFolders returns org folders by their UIDs.
+// If UIDs is empty, it returns all folders in the org.
+// If WithFullpath is true it computes also the full path of a folder.
+// The full path is a string that contains the titles of all parent folders separated by a slash.
+// For example, if the folder structure is:
+//
+//	A
+//	└── B
+//	    └── C
+//
+// The full path of C is "A/B/C".
+// The full path of B is "A/B".
+// The full path of A is "A".
+// If a folder contains a slash in its title, it is escaped with a backslash.
+// For example, if the folder structure is:
+//
+//	A
+//	└── B/C
+//
+// The full path of C is "A/B\/C".
+func (ss *sqlStore) GetFolders(ctx context.Context, q getFoldersQuery) ([]*folder.Folder, error) {
+	if q.BatchSize == 0 {
+		q.BatchSize = DEFAULT_BATCH_SIZE
+	}
+
+	if len(q.ancestorUIDs) == 0 && len(q.UIDs) == 0 {
+		return nil, folder.ErrBadRequest.Errorf("one of UIDs or AncestorUIDs must be included in the query")
```

Comment:
Does this mean that if a user that has wildcard permissions to folders will not be able to request all folders, right?

Reply:
yes but I have removed it